apis/gitops.turbonomic.io/v1alpha1/gitopsmode_types.go

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// GitOpsModeSpec defines the desired state of GitOpsMode
+type GitOpsModeSpec struct {
+	// Overrides the default GitOps configuration with custom configuration for the specified app(s).
+	Configutaion []GitOpsConfiguration `json:"configuration"`
+}
+
+// CommitMode describes how Turbonomic events will be processed.
+// +kubebuilder:validation:Enum=direct;pr
+type CommitMode string
+
+const (
+	// Actions will produce commit directly within the underlying repository without creating a pull/merge request
+	DirectCommit CommitMode = "direct"
+
+	// Actions will result in a pull/merge request being creating within the underlying repository
+	PullRequest CommitMode = "pr"
+)
+
+type GitOpsConfiguration struct {
+	// Specifies the GitOps commit mode.
+	// Valid values are:
+	// - "direct": actions will produce commit directly within the underlying repository without creating a pull/merge request;
+	// - "pr": actions will result in a pull/merge request being creating within the underlying repository
+	CommitMode CommitMode `json:"commitMode"`
+	// Specifies the credentials for the underlying repository (CURRENTLY UNSUPPORTED)
+	// +optional
+	Credentials GitOpsCredentials `json:"credentials,omitempty"`
+	// Specifies the applications that the commit mode should apply to
+	Applications []string `json:"applications"`
```

Comment:
Shall we add a validation here to make sure the list of applications is **NOT** empty?

Reply:
Sure. I'll get that added.

Reply:
Added validations to ensure list of apps is nonempty.